File: frontend/src/types/feedback.ts
export interface SubmitFeedbackRequest {
  query_id: string;
  feedback_type: string;
  citation_id?: string;
  description?: string;
  submitted_by?: string;
}

export interface FeedbackDetail {
  id: string;
  query_id: string;
  feedback_type: string;
  citation_id: string | null;
  description: string;
  submitted_by: string;
  status: string;
  resolution: string | null;
  traced_source_id: string | null;
  traced_manifest_id: string | null;
  traced_document_id: string | null;
  auto_action: string | null;
  submitted_at: string;
  resolved_at: string | null;
}

export interface ResolveFeedbackRequest {
  resolution: string;
  status?: string;
}

export interface CurationQueueItem {
  id: string;
  source_id: string;
  manifest_id: string;
  priority: string;
  reason: string;
  trigger_type: string;
  feedback_id: string | null;
  change_event_id: string | null;
  status: string;
  result: string | null;
  created_at: string;
  processed_at: string | null;
}

export interface ChangeEvent {
  id: string;
  source_id: string;
  manifest_id: string;
  detection_method: string;
  change_type: string;
  previous_hash: string | null;
  current_hash: string | null;
  description: string;
  status: string;
  impact_assessment: string | null;
  detected_at: string;
  resolved_at: string | null;
}

export interface TriggerMonitorResponse {
  sources_checked: number;
  changes_detected: number;
  message: string;
}

export interface AccuracyMetrics {
  total_feedback: number;
  correct_count: number;
  inaccurate_count: number;
  outdated_count: number;
  incomplete_count: number;
  irrelevant_count: number;
  accuracy_score: number;
  resolution_rate: number;
  avg_source_confidence: number;
  stale_sources: number;
  pending_queue_items: number;
  unresolved_changes: number;
}

export interface AccuracyTrendPoint {
  date: string;
  accuracy_score: number;
  total_feedback: number;
  resolution_rate: number;
}

export interface AccuracyDashboardData {
  current: AccuracyMetrics;
  trends: AccuracyTrendPoint[];
  by_feedback_type: Record<string, number>;
  by_vertical: Record<string, number>;
}

export const FEEDBACK_TYPES = [
  { value: "inaccurate", label: "Inaccurate", color: "#ef4444" },
  { value: "outdated", label: "Outdated", color: "#eab308" },
  { value: "incomplete", label: "Incomplete", color: "#f97316" },
  { value: "irrelevant", label: "Irrelevant", color: "#8b5cf6" },
  { value: "correct", label: "Correct", color: "#22c55e" },
] as const;

export const PRIORITY_COLORS: Record<string, string> = {
  critical: "#ef4444",
  high: "#f97316",
  medium: "#eab308",
  low: "#3b82f6",
};

export const STATUS_COLORS: Record<string, string> = {
  pending: "#eab308",
  investigating: "#3b82f6",
  processing: "#3b82f6",
  resolved: "#22c55e",
  dismissed: "#6b7280",
  detected: "#ef4444",
};

File: backend/app/schemas/feedback.py
"""Pydantic schemas for feedback, re-curation queue, change monitoring, accuracy."""

from datetime import datetime

from pydantic import BaseModel, Field

# --- Feedback ---

class SubmitFeedbackRequest(BaseModel):
    query_id: str
    feedback_type: str = Field(
        description="inaccurate | outdated | incomplete | irrelevant | correct"
    )
    citation_id: str | None = None
    description: str = ""
    submitted_by: str = "anonymous"


class FeedbackDetail(BaseModel):
    id: str
    query_id: str
    feedback_type: str
    citation_id: str | None
    description: str
    submitted_by: str
    status: str
    resolution: str | None
    traced_source_id: str | None
    traced_manifest_id: str | None
    traced_document_id: str | None
    auto_action: str | None
    submitted_at: datetime
    resolved_at: datetime | None


class ResolveFeedbackRequest(BaseModel):
    resolution: str
    status: str = "resolved"  # resolved | dismissed


# --- Curation Queue ---

class CurationQueueItemSchema(BaseModel):
    id: str
    source_id: str
    manifest_id: str
    priority: str
    reason: str
    trigger_type: str
    feedback_id: str | None
    change_event_id: str | None
    status: str
    result: str | None
    created_at: datetime
    processed_at: datetime | None


# --- Change Events ---

class ChangeEventSchema(BaseModel):
    id: str
    source_id: str
    manifest_id: str
    detection_method: str
    change_type: str
    previous_hash: str | None
    current_hash: str | None
    description: str
    status: str
    impact_assessment: str | None
    detected_at: datetime
    resolved_at: datetime | None


class TriggerMonitorResponse(BaseModel):
    sources_checked: int
    changes_detected: int
    message: str


# --- Accuracy Dashboard ---

class AccuracyMetrics(BaseModel):
    total_feedback: int
    correct_count: int
    inaccurate_count: int
    outdated_count: int
    incomplete_count: int
    irrelevant_count: int
    accuracy_score: float
    resolution_rate: float
    avg_source_confidence: float
    stale_sources: int
    pending_queue_items: int
    unresolved_changes: int


class AccuracyTrendPoint(BaseModel):
    date: str
    accuracy_score: float
    total_feedback: int
    resolution_rate: float


class AccuracyDashboardData(BaseModel):
    current: AccuracyMetrics
    trends: list[AccuracyTrendPoint]
    by_feedback_type: dict[str, int]
    by_vertical: dict[str, float]
